pkg/apiserver/resthandler.go

```diff
@@ -269,8 +269,14 @@ func CreateResource(r rest.Creater, scope RequestScope, typer runtime.ObjectType
 			errorJSON(err, scope.Codec, w)
 			return
 		}
+		// Get name, ok if empty
+		_, name, _ := scope.Namer.Name(req)
+
 		ctx := scope.ContextFunc(req)
 		ctx = api.WithNamespace(ctx, namespace)
+		if len(name) > 0 {
+			ctx = api.WithValue(ctx, "name", name)
```

Comment:
More like, if you implemented an optional interface, you could set "name" directly into the runtime.Object we pass to create.  Then it's your caller's responsibility to use that name somehow.  Examples might be for parent resource, or bindings, or just populating a field.

----- Original Message -----
> >  		ctx := scope.ContextFunc(req)
> >  		ctx = api.WithNamespace(ctx, namespace)
> > +		if len(name) > 0 {
> > +			ctx = api.WithValue(ctx, "name", name)
> 
> Post/Create is kind of unique because it doesn't take a name parameter. Not
> sure what you mean by adding it to the other objects. It would be like
> GetterWithOptions... if your storage object happens to implement it, we use
> it, if not, then great.
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/GoogleCloudPlatform/kubernetes/pull/7718/files#r29616289